core/parser/parser_test.go

```diff
@@ -2,58 +2,252 @@ package parser_test
 
 import (
 	"github.com/otrego/clamshell/core/parser"
+	"github.com/otrego/clamshell/core/game"
+	"reflect"
 	"testing"
+    "os"
 )
 
-func TestCreate(t *testing.T) {
-	s := `
-(;GM[1]FF[4]CA[UTF-8]AP[CGoban:3]ST[2]
-RU[Japanese]SZ[19]KM[6.50]
-PW[Player White [1k\]]PB[Player Black [3k\]]C[This (yes;) is a [5k\] comment]
-;AW[jj][ki]
-;B[pd]C[Watcher [10k\]: hello world]
-;W[dd]
-;B[pq]
-;W[dp]
-(;B[qk]
-(;W[fq]
-;B[nc]
-;W[fc]
-;B[cj]
-;W[cl]
-;B[cf]
-;W[mp]
-;B[po]
-;W[jp]
-;B[ej])
-(;W[nc]
-;B[pf]
-;W[pb]
-(;B[qc]
-;W[kc])
-(;B[fq]))
-(;W[mp]
-;B[po]
-;W[jp]))
-(;B[fq]
-(;W[cn]
-;B[jp])
-(;W[qo])))`
-	p := parser.FromString(s)
-	game, err := p.Parse()
+type PropMap map[string][]string
+
+func testFile(dir, fname string, exp map[game.Path]PropMap, t *testing.T) {
+    fd,err := os.Open(dir + "/" + fname)
+    if err != nil {
+        t.Error(err)
+    }
+
+    stat,err := fd.Stat()
+    if err != nil {
+        t.Error(err)
+    }
+
+    size := stat.Size()
+    data := make([]byte, size)
+
+    _,err = fd.Read(data)
+    if err != nil {
+        t.Error(err)
+    }
+
+	p := parser.FromString(string(data))
+	g, err := p.Parse()
 	if err != nil {
 		t.Error(err)
 	}
-	node := game.Root.Children[0]
 
-	if pw := node.Properties["PW"][0]; pw != "Player White [1k\\]" {
-		t.Errorf("PW=%v, expected %v", pw, "Player White [1k\\]")
-	}
-	if pb := node.Properties["PB"][0]; pb != "Player Black [3k\\]" {
-		t.Errorf("PB=%v, expected %v", pb, "Player Black [3k\\]")
-	}
-	nxt := node.Children[0]
-	if aw := nxt.Properties["AW"]; len(aw) < 2 || aw[0] != "jj" || aw[1] != "ki" {
-		t.Errorf("AW=%v, expected %v", aw, []string{"jj", "ki"})
-	}
+    for path, propMap := range exp {
+        node, err := g.TreePath(path)
+        if err != nil {
+            t.Error(err)
+            continue
+        }
+
+        for prop, values := range propMap {
+            if chk := node.Properties[prop]; !reflect.DeepEqual(values, chk) {
+                t.Errorf("%s: %s=%v, expected %v", fname, prop, node.Properties[prop], values)
+            }
+
+        }
+    }
+
+}
+
+func TestBase(t *testing.T) {
+    dir := "testdata"
+    fname := "base.sgf"
+    expectedProperties := map[game.Path]PropMap{
+        "0": map[string][]string{
+            "GM": []string{"1"},
+        },
+    }
+    testFile(dir, fname, expectedProperties, t)
+}
+
+func TestDescription(t *testing.T) {
```

Comment:
Can you make these into table tests rather than individual test methods? https://github.com/golang/go/wiki/TableDrivenTests 

That would be more idiomatic